### PR TITLE
Service

### PR DESCRIPTION
**Pull request by trisrael**

### CODE REVIEW

**Comment by trisrael** [comment]
@alexgb needed for 16.0.4 release.

File: addon/services/segmentio.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  'once',
+  'off',
+  'on'
+];
+
+export default Ember.Service.extend({
+  init() {
+    if (!window.analytics) { return; }
+    SEGMENT_METHODS.forEach((methodName) => {
+      this[methodName] = window.analytics[methodName];
```
Comment (alexgb) [inline addon/services/segmentio.js]: Isn't this potentially buggy because you're not binding? You may want to do:

``` javascript
this[methodName] = window.analytics[methodName].bind(window.analytics);
```

Comment (trisrael) [inline addon/services/segmentio.js]: yah i thought about that. Okay will do

File: index.js [hunk 2]
```diff
@@ -38,11 +38,11 @@ module.exports = {
     if (enabled && type === 'head') {
       content = [
         '<script type="text/javascript">',
-        '!function(){',
+        '(function(){',
```
Comment (alexgb) [inline index.js:41]: Why this change?

File: addon/services/segmentio.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import Ember from 'ember';
+
+// Should match methods in snippet-core.min.js
+const SEGMENT_METHODS = [
```
Comment (alexgb) [inline addon/services/segmentio.js]: Can we just iterate through all methods of `window.analytics`?

File: index.js (hunk 2 above)
Comment (trisrael) [inline index.js:41]: was blowing up on this line in the script

File: addon/services/segmentio.js (hunk 3 above)
Comment (trisrael) [inline addon/services/segmentio.js]: Am doing that now 49 properties so not terrible; however [stupid phantomjs](https://groups.google.com/forum/#!msg/phantomjs/r0hPOmnCUpc/uxusqsl2LNoJ)

File: addon/services/segmentio.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import Ember from 'ember';
+
+export default Ember.Service.extend({
+  init() {
+    if (!window.analytics) { return; }
+
+    // copy all functions from analytics
+    Object.getOwnPropertyNames(window.analytics)
+      .forEach((methodName) => {
+        console.count('methods');
```
Comment (trisrael) [inline addon/services/segmentio.js]: remove

**Comment by trisrael** [comment]
@alexgb okay ready for another round
